### PR TITLE
docs: add docs for dump reload

### PR DESCRIPTION
**Pull request by cristianmtr**
Documentation for dump - reload, query while indexing, replicas

### All Submissions:

* [x] Have you followed the [documentation style guide](https://github.com/jina-ai/docs/tree/master/page_templates/style_guide.md)? 

### New Feature Submissions:

1. [x] Have you added your new file to a table of content index? 
2. [x] Did you use one of the templates in the [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) folder?
3. [x] Is your title in Title Case? 
4. [x] Are your section headers in sentence case?

### CODE REVIEW

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: In this section, you will learn how to configure Jina to allow **queries** while indexing.

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: **Querying** while 

data is **simultaneously** being written to it.

File: chapters/dump-reload.md
Comment (maximilianwerk) [inline chapters/dump-reload.md]:
```suggestion
`.rolling_update` should be called on the Query Flow, with the name of the Pod where the Indexer resides, and the path from which to read.
```

File: chapters/dump-reload.md
Comment (maximilianwerk) [inline chapters/dump-reload.md]:
```suggestion
The two systems communicate with each other using a system of rolling update calls to the respective Flows. 
```

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: paradigm or feature? I think I would market this as a feature.

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: This **allows you to** have

File: chapters/dump-reload.md
Comment (FionnD) [reply chapters/dump-reload.md]: if you choose to use feature instead of paradigm, then check the whole document and update paradigm

**Review by maximilianwerk — DISMISSED** [review]
I would love to see some kind of graph illustrating the dump/reload mechanism. We had several during the design phase and can perhaps recycle one with only minor changes?

Apart from that, one idea: Should we have two more sections "the replica parameter" and "the shards parameter", which explain these two parameters. This would allow a consistent linking whenever these parameters are used elsewhere in our documentation.

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: 1) As a stupid user, it would confuse me to mentally try and understand what a DBMS indexer is, and then compare it to a CRUD-like interface? That's a lot of abbreviations to hold in memory. 

I tried to rephrase it, but might not be any clearer. I'll leave it up to you to accept this suggestion or not. 

The **DBMS** (Database Management System) indexers store both the Document's embedding and the Document's metadata as one entity.  They also provide the `index`, `update`, and `delete` methods for this entity.

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: file system-based storage where? They dump to your workspace right?

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: called

Comment (FionnD) [reply chapters/dump-reload.md]: Should there be any code snippets for this behaviour?

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: DBMS (index) Flow

This make it clearer for the use who may not understand that the DBMS Flow replaces your index flow

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: JinaD

**Comment by cristianmtr** [comment]
> I would love to see some kind of graph illustrating the dump/reload mechanism. We had several during the design phase and can perhaps recycle one with only minor changes?

@FionnD will have some input on this. The original ones weren't very clean.

> Apart from that, one idea: Should we have two more sections "the replica parameter" and "the shards parameter", which explain these two parameters. This would allow a consistent linking whenever these parameters are used elsewhere in our documentation.

Expanded on the replica param. The shards param is covered someplace else. I've added a link to it

File: chapters/dump-reload.md
Comment (FionnD) [inline chapters/dump-reload.md]: Suggestion: Add this to make it clearer to the user how the next three section interact. 



In the subsections below, we explain the following three concepts. 

(Add this as a table)

Section name - Explantation 
 DBMS and Query Indexers - Understanding the two Flows you will need to set up 
 Dump-Reload - The configuration you will need to ensure data passes between the Flows 
 Replicas  - The configuration you will need to uptime as data passes between Flows.

**Review by FionnD — CHANGES_REQUESTED** [review]
Really good, some suggestions

**Comment by cristianmtr** [comment]
@FionnD I implemented your suggestions

**Review by FionnD — APPROVED** [review]
LGTM to me, just need to decide about the graphics?

File: chapters/dump-reload.md
Comment (CatStark) [inline chapters/dump-reload.md]: Can you change this to Title Case?

File: chapters/dump-reload.md
Comment (CatStark) [inline chapters/dump-reload.md]: This should be Sentence case

File: chapters/dump-reload.md
Comment (CatStark) [inline chapters/dump-reload.md]: Also sentence case

**Review by CatStark — DISMISSED** [review]

File: chapters/dump-reload.md
Comment (cristianmtr) [reply chapters/dump-reload.md]: Query Indexers are a class type

File: chapters/dump-reload.md
Comment (cristianmtr) [reply chapters/dump-reload.md]: done

File: chapters/dump-reload.md
Comment (cristianmtr) [reply chapters/dump-reload.md]: done

**Comment by FionnD** [comment]
We also need graphics before this is merged, either the ones I slacked you. Or mock up your own.

File: chapters/dump-reload.rst
Comment (FionnD) [inline chapters/dump-reload.rst]: Full stop.

File: chapters/dump-reload.rst
Comment (FionnD) [inline chapters/dump-reload.rst:121]: Full stop.

File: chapters/dump-reload.rst
Comment (FionnD) [inline chapters/dump-reload.rst:24]: add example

File: chapters/dump-reload.rst
Comment (FionnD) [inline chapters/dump-reload.rst:31]: these images are huge.  You should be able to adjust them to something more reasonable? 

Screenshot 2021-05-17 at 13 41 02

**Review by FionnD — CHANGES_REQUESTED** [review]

**Review by FionnD — APPROVED** [review]
LGTM!

File: chapters/dump-reload.rst
Comment (alexcg1) [inline chapters/dump-reload.rst]: - are either

File: chapters/dump-reload.rst
Comment (alexcg1) [inline chapters/dump-reload.rst:39]: Are we capitalizing Index and Query @FionnD ?

File: chapters/dump-reload.rst
Comment (alexcg1) [inline chapters/dump-reload.rst]: @FionnD is this a separate example or will we merge it into the wiki example like we did incremental indexing?

**Review by alexcg1 — CHANGES_REQUESTED** [review]

**Comment by alexcg1** [comment]
Great docs post! One tiny tweak and a couple of questions for @FionnD

File: chapters/dump-reload.rst
Comment (FionnD) [reply chapters/dump-reload.rst]: Separate as per discussion here https://jinaai.slack.com/archives/C018NC4SYAZ/p1620801607122900

File: chapters/dump-reload.rst
Comment (FionnD) [reply chapters/dump-reload.rst:39]: Good catch, yes we should because they are in the https://docs.jina.ai/chapters/glossary/ 

**Rule: Anything in the glossary is caplitzed**

File: chapters/dump-reload.rst
Comment (cristianmtr) [reply chapters/dump-reload.rst]: We won't, since it's radically different. The incr indexing is v similar.

**Comment by cristianmtr** [comment]
> Great docs post! One tiny tweak and a couple of questions for @FionnD

Addressed

**Review by alexcg1 — COMMENTED** [review]
LGTM :+1:

**Comment by cristianmtr** [comment]
> LGTM

I think you need to explicitly approve

**Review by alexcg1 — APPROVED** [review]

**Comment by alexcg1** [comment]
> > LGTM
> 
> I think you need to explicitly approve

Whoops. Done!

**Pull request merged**
